lib/Basics/Exceptions.cpp

```diff
@@ -170,3 +154,17 @@ std::string Exception::FillFormatExceptionString(char const* format, ...) {
       << "Failed to create an error message, giving up. " << errorMessage;
   FATAL_ERROR_EXIT();
 }
+
+[[noreturn]] void ::arangodb::basics::abortOrThrowException(Exception e) {
```

Comment:
I guess `e` should be reference here, so we don't copy it.
```suggestion
[[noreturn]] void ::arangodb::basics::abortOrThrowException(Exception const& e) {
```

Reply:
`throw` copy-initializes the exception. So my train of thought was, if we take a `const&` here, that'd mean we'd have to copy (instead of moving) it when throwing, unconditionally. This way, the caller can at least choose to move the exception to avoid a copy altogether. However, I seem to have forgotten the move in non-maintainer mode; I will fix that.

Reply:
Taking the exception by value might also result in object slicing if we have a subtype of Exception. How about making this a template and the argument a universal reference?

Reply:
Good point!